addons/sale/models/sale_order_line.py

```diff
@@ -377,7 +377,7 @@ def _compute_no_variant_attribute_values(self):
     @api.depends('product_id', 'linked_line_id', 'linked_line_ids')
     def _compute_name(self):
         for line in self:
-            if not line.product_id and not line.is_downpayment:
+            if (not line.product_id and not line.is_downpayment) or not line.order_id:
```

Comment:
```suggestion
            if not (line.order_id and (line.product_id or line.is_downpayment)):
```
I think this would be easier to read than having three negations in three clauses.